### DISCUSSION TITLE
Issues when distinguishing between sources

### DISCUSSION THREAD

**Discussion by KiyoakiOmori** [sep-developers/sep#131] [Q&A]
I am running the sep package so I can identify sources for the segmentation map for the purpose of running statmorph, and ran into an odd issue.

When trying to identify sources/segments, the division between sources seems to be highly dependent on the thresh parameter, instead of the deblend_nthresh and deblend_cont values. I attach an example. In the thresh1.5 image, there are multiple sources with a clear division in the center, but they are shown as one source in the thresh1.0 image. Changing the deblend_nthresh and deblend_cont values seemingly did not have an effect on distinguishing the sources.

Threshold 1.5 image
![thresh1 5](https://user-images.githubusercontent.com/50357063/139790026-6240d34e-4456-4f67-990c-e94c03f2afc4.png)

Threshold 1.0 image
![thresh1 0](https://user-images.githubusercontent.com/50357063/139790033-0b28a442-9280-44f8-a6c5-bfae2a5e99d2.png)

**Comment by kbarbary** [comment]
At first glance, it isn't surprising that `thresh` would have this effect. A lower `thresh` means more pixels are over the threshold. In regions of low surface brightness extended objects, these additional "detected" pixels will connect otherwise separated high points, forming larger objects.

I am a little surprised that in the "thresh1.5" image, the largest blob seems to have been deblended into 3 objects, but not in the `thresh1.0` image. You may want to try entirely disabling deblending via `deblend_cont=1.0` to see if this is really even due to deblending. Turning off deblending can also help you disentangle the effects of changing the threshold and the effects of the deblending algorithm.

As for why `deblend_nthresh` and `deblend_cont` seem to have no effect: I'm not entirely sure. It might depend on what the high points of these objects look like, as the deblending algorithm is designed to separate overlapping objects with clearly separated peaks. If there aren't clearly separated peaks, I wouldn't be too surprised that deblending doesn't do much with any setting.
